src/app.ts

```diff
@@ -1,18 +1,28 @@
 import express from 'express';
+import cors, { CorsOptions } from 'cors';
 
 import { usersRouter, groupsRouter } from './controllers/index';
 import { requestLogger, logger } from './logging/index';
-import { errorHandler } from './middlewares/index';
+import { authentication, errorHandler, checkToken } from './middlewares/index';
+
+const corsOptions: CorsOptions = {
+    origin: '*',
+    methods: ['GET', 'HEAD', 'PUT', 'POST', 'DELETE'],
+    optionsSuccessStatus: 200
+};
 
 const app: express.Express = express();
 const port: string = process.env.PORT || '3000';
 
+app.use(cors(corsOptions));
 app.use(express.json());
 
 app.use(requestLogger);
 
-app.use('/users', usersRouter);
-app.use('/groups', groupsRouter);
+app.post('/login', authentication);
```

Comment:
actually its not middleware, it's better to create separate auth controller, as in advanced case there also could be such methods as refreshToken, logOut.
And would be nice to move logic about token signing and decryption to separate JwtService